lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1206,7 +1206,8 @@ def _build_files_manifest_walk(b_collection_path, namespace, name, ignore_patter
     def _walk(b_path, b_top_level_dir):
         for b_item in os.listdir(b_path):
             b_abs_path = os.path.join(b_path, b_item)
-            b_rel_base_dir = b'' if b_path == b_top_level_dir else b_path[len(b_top_level_dir) + 1:]
+            common_prefix = os.path.commonprefix([b_top_level_dir, b_path])
+            b_rel_base_dir = b_path[len(common_prefix):].lstrip(b'/')
```

Comment:
[todo (blocking)][conventional comments]: I'd like to request that `os.path.sep` is used instead of the hard-coded separator.

[conventional comments]: https://conventionalcomments.org/

Reply:
Ahh this is important to allow this to work in both platforms (Linux/Windows) where the separator is actually different. Thanks for this.

Reply:
Well, technically we don't support running on Windows natively. But WSL2 should work. I just try to avoid intentional obstacles in case we ever start supporting it...

Reply:
Marking this thread as resolved.